service/resource/endpoint/update.go

```diff
@@ -3,32 +3,42 @@ package endpoint
 import (
 	"context"
 
+	"k8s.io/apimachinery/pkg/api/errors"
+
 	"github.com/giantswarm/microerror"
 	"github.com/giantswarm/operatorkit/framework"
 )
 
-func (r *Resource) NewUpdatePatch(ctx context.Context, obj, currentState, desiredState interface{}) (*framework.Patch, error) {
-	create, err := r.newCreateChange(ctx, obj, currentState, desiredState)
+func (r *Resource) ApplyUpdateChange(ctx context.Context, obj, updateState interface{}) error {
+	k8sEndpoint, err := toK8sEndpoint(updateState)
 	if err != nil {
-		return nil, microerror.Mask(err)
+		return microerror.Mask(err)
+	}
+
+	if isEmptyEndpoint(*k8sEndpoint) {
+		return nil
+	}
+
+	_, err = r.k8sClient.CoreV1().Endpoints(k8sEndpoint.Namespace).Update(k8sEndpoint)
+	if errors.IsNotFound(err) {
+		return nil
+	} else if err != nil {
+		return microerror.Mask(err)
 	}
 
-	update, err := r.newUpdateChange(ctx, obj, currentState, desiredState)
+	return nil
+}
+
+func (r *Resource) NewUpdatePatch(ctx context.Context, obj, currentState, desiredState interface{}) (*framework.Patch, error) {
+	k8sEndpoint, err := r.newCreateChange(ctx, obj, currentState, desiredState)
 	if err != nil {
 		return nil, microerror.Mask(err)
 	}
 
 	patch := framework.NewPatch()
-	patch.SetCreateChange(create)
-	patch.SetUpdateChange(update)
 
-	return patch, nil
-}
+	patch.SetCreateChange(k8sEndpoint)
+	patch.SetUpdateChange(k8sEndpoint)
```

Comment:
What is this? I do not think it makes a lot of sense to set the create and the update patch at the same time with the same state.

Reply:
I changed this to have two different states now, similar to what I saw in cluster-operator.